helix-lock/src/test/java/org/apache/helix/lock/helix/TestZKHelixNonblockingLock.java

```diff
@@ -71,9 +73,14 @@ public void beforeMethod() {
     Assert.assertFalse(_gZkClient.exists(_lockPath));
   }
 
+  @AfterSuite
+  public void afterSuite() throws IOException {
+    _lock.close();
```

Comment:
Add a test case about close() to validate:
1. lock released properly. The releases information should be right (if we have any).
2. the connection has been ended.

Reply:
I added two tests. One is trying to close the lock when it's unlocked, and one is trying to close it when it's locked.